### ISSUE TITLE
Rocket on uswgi

### ISSUE THREAD

**Issue by lu-zero** [SergioBenitez/Rocket#97]
[uswgi](https://github.com/unbit/uwsgi-rust) had already some experimental rust support and they were looking for feedback about the interface in use.

Would be possible to wire rocket in uswgi modifying it? how?

**Comment by SergioBenitez** [comment]
I think it would be pretty simple to get a Rocket app running under any kind of CGI by abusing the [testing module](https://api.rocket.rs/rocket/testing/). You can simply:

  1. Construct a `MockRequest` from the environment.
  2. Dispatch it.
  3. Write the `Response` out.

CGI doesn't fit will with Rocket's goals, so I don't see it being part of core. However, this would be a nice stand-alone library, or if it proves useful enough, part of `contrib`.

**Comment by SergioBenitez** [comment]
Closing this out. Feel free to reopen if you feel this is unanswered.

**Comment by lu-zero** [comment]
Not sure if would be better to document somewhere now or wait later and have a deploy guide once the dust is settled regarding what's the underlying http library.

**Comment by SergioBenitez** [comment]
@lu-zero I'm sorry, can you rephrase that? I'm not sure I understand.

**Comment by lu-zero** [comment]
I meant that might be a good idea to put somewhere in the documentation how to wire rocket to some kind of cgi or such. I assume people would have interest in fitting rocket in the existing infrastructure they already have.

Probably it isn't the top concern now given the discussions about using tokio, but one that part is settled it might be quite useful.
